src/contract/HandlerBasedContract.ts

```diff
@@ -224,13 +234,21 @@ export class HandlerBasedContract<State> implements Contract<State> {
     return interactionTx.id;
   }
 
-  async bundleInteraction<Input>(input: Input, tags: Tags = [], strict = false): Promise<any | null> {
+  async bundleInteraction<Input>(input: Input, tags: Tags = [], strict = false): Promise<any> {
     this.logger.info('Bundle interaction input', input);
     if (!this.wallet) {
       throw new Error("Wallet not connected. Use 'connect' method first.");
     }
-    const interactionTx = await this.createInteraction(input, tags, emptyTransfer, strict);
+    const interactionTx = await this.createUnsignedInteraction(input, tags, emptyTransfer, strict);
+    await this.smartweave.arweave.transactions.sign(interactionTx, this.wallet);
+
+    return this.bundleInteractionTx(interactionTx);
+  }
+
+  async bundleInteractionTx(interactionTx: Transaction): Promise<any> {
+    this.smartweave.arweave.transactions.verify(interactionTx);
```

Comment:
I believe this call is missing `await`

Reply:
also..the result doesn't seem to be used anywhere - I believe the idea was to throw if the `verify` return `false`?